### DISCUSSION TITLE
Usage of versionCompatibility for Maven dependencies?

### DISCUSSION THREAD

**Discussion by TobiX** [renovatebot/renovate#27457] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.170.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

I tried using [`versionCompatibility`](https://docs.renovatebot.com/configuration-options/#versioncompatibility) for the Microsoft JDBC driver (https://github.com/Microsoft/mssql-jdbc) which is released for different JDKs and an update shouldn't change that.

I tried this:

```json
  "packageRules": [
    {
      "description": "Honor MSSQL JDBC compatibility tag",
      "matchDatasources": ["maven"],
      "matchPackageNames": ["com.microsoft.sqlserver:mssql-jdbc"],
      "versionCompatibility": "^(?<version>.*)(?<compatibility>\\.jre.*)$"
    }
  ]
```

The update was not what I expected: Renovates removes the "compatibility tag" completely instead of leaving it as it is...

See also the PR in my reproducer repository: https://github.com/TobiX/test-renovate-stuff/pull/6

### Logs (if relevant)

<details><summary>Logs</summary>

```json
          {
            "currentValue": "12.4.2.jre8",
            "currentVersion": "12.4.2",
            "datasource": "maven",
            "depName": "com.microsoft.sqlserver:mssql-jdbc",
            "depType": "compile",
            "fileReplacePosition": 1010,
            "fixedVersion": "12.4.2",
            "isSingleVersion": true,
            "packageName": "com.microsoft.sqlserver:mssql-jdbc",
            "registryUrl": "https://repo.maven.apache.org/maven2",
            "registryUrls": [
              "https://repo.maven.apache.org/maven2"
            ],
            "sourceUrl": "https://github.com/Microsoft/mssql-jdbc",
            "versioning": "maven",
            "warnings": [],
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "12.6.0",
                "newValue": "12.6.0",
                "releaseTimestamp": "2024-02-01T19:02:30.000Z",
                "newMajor": 12,
                "newMinor": 6,
                "updateType": "minor",
                "branchName": "renovate/com.microsoft.sqlserver-mssql-jdbc-12.x"
              }
            ]
          }
```

</details>

**Comment by rarkins** [comment]
This is not what I'd expect to happen. I'll try to look into it soon

**Reply by rarkins** [reply, marked as answer]
Fix PR: https://github.com/renovatebot/renovate/pull/27490
